src/Orleans/Async/TaskExtensions.cs

```diff
@@ -51,6 +53,62 @@ public static class PublicOrleansTaskExtentions
                 var ignored = task.Exception; // Observe exception
             }
         }
+
+        /// <summary>
+        /// Returns a <see cref="Task{Object}"/> for the provided <see cref="Task"/>.
+        /// </summary>
+        /// <param name="task">
+        /// The task.
+        /// </param>
+        /// <returns>
+        /// The response.
+        /// </returns>
+        public static Task<object> Box(this Task task)
+        {
+            return task.ContinueWith(
+                antecedent =>
+                {
+                    if (antecedent.Exception != null)
+                    {
+                        throw new AggregateException(antecedent.Exception);
```

Comment:
Why is the exception handling logic different between the two versions of `Box()`?